### PR TITLE
Change not auto fleet scouts and cols

### PR DESCRIPTION
**Pull request by LGM-Doyle**
Please use the following as the merge message.

---------------------snip --------------------------------------
1. Names fleets based on their function: Colony, Recon, Troop, Bomber, Combat and Herd.
2. Prevent non-combat fleets from automatically merging.  Colony and Recon ships are immediately useful for sending to new planets.

### CODE REVIEW

**Comment by Vezzra** [comment]
For release or post release?

**Comment by LGM-Doyle** [comment]
I would include it in the release because:
- It is purely an minor UI improvement.
- It does not change any rules and hence does not affect balance.
- Play testing of the release candidate has not started yet.

This PR prevents non-combatant, scouts, colony and outpost ships from automatically forming fleets when they are produced in the same turn at the same planet.  

The changes are useful by turn 3 with high production races like the Egassem.  

If an Egassem empire adds 2 scouts to the production queue on turn 1, on turn 3 they produce 2 scouts.  Currently, the scouts group into a fleet, which is not what the player wants.  The player wants 2 scouts to head off in two different directions and find an Inferno planet.

**Comment by MatGB** [comment]
I'd agree with this, it's not something I ever thought to ask for but as soon as I saw the description I realised it was something I definitely wanted. I want my warships grouped, I don't want my scouts or colonies grouped _unless_ I've told them to go to the same place with a rally point, and even that's not necessary.

File: universe/Fleet.cpp [hunk 1]
```diff
@@ -1391,13 +1391,101 @@ void Fleet::ShortenRouteToEndAtSystem(std::list<int>& travel_route, int last_sys
...
+        if (TemporaryPtr<const Ship> ship = GetShip(*it)) {
+            ships.push_back(ship);
+        }
+    }
+
+    std::vector< TemporaryPtr<const Ship> >::iterator it = ships.begin();
+
+    // TODO C++11 replace all of these loops with std::all_of
+    bool all_monster(true);
+    while (it != ships.end()) {
```
Comment (geoffthemedio) [inline universe/Fleet.cpp:1411]: why while and not for?

File: universe/Fleet.cpp [hunk 2]
```diff
@@ -1391,13 +1391,101 @@ void Fleet::ShortenRouteToEndAtSystem(std::list<int>& travel_route, int last_sys
...
+        if(!(*it)->IsMonster()) {
```
Comment (geoffthemedio) [inline universe/Fleet.cpp]: space after if

File: universe/Fleet.cpp [hunk 3]
```diff
@@ -1391,13 +1391,101 @@ void Fleet::ShortenRouteToEndAtSystem(std::list<int>& travel_route, int last_sys
...
 
-    return boost::io::str(FlexibleFormat(UserString("NEW_FLEET_NAME")) % boost::lexical_cast<std::string>(new_fleet_id));
+    std::vector<TemporaryPtr<const Ship> > ships;
+    for (std::set<int>::iterator it = m_ships.begin(); it != m_ships.end(); ++it) {
+        if (TemporaryPtr<const Ship> ship = GetShip(*it)) {
+            ships.push_back(ship);
+        }
+    }
+
+    std::vector< TemporaryPtr<const Ship> >::iterator it = ships.begin();
```
Comment (geoffthemedio) [inline universe/Fleet.cpp]: remove extra space after <

File: universe/Fleet.cpp [hunk 4]
```diff
@@ -1391,13 +1391,101 @@ void Fleet::ShortenRouteToEndAtSystem(std::list<int>& travel_route, int last_sys
...
+        ++it;
+    }
+
+    if (all_non_coms)
+        return boost::io::str(FlexibleFormat(UserString("NEW_RECON_FLEET_NAME")) % boost::lexical_cast<std::string>(ID()));
+
+    bool all_troop(true);
+    it = ships.begin();
+    while (it != ships.end()) {
+        if(!(*it)->CanHaveTroops()) {
```
Comment (geoffthemedio) [inline universe/Fleet.cpp]: space after if

File: universe/Fleet.cpp [hunk 5]
```diff
@@ -1391,13 +1391,101 @@ void Fleet::ShortenRouteToEndAtSystem(std::list<int>& travel_route, int last_sys
...
+        ++it;
+    }
+
+    if (all_troop)
+        return boost::io::str(FlexibleFormat(UserString("NEW_TROOP_FLEET_NAME")) % boost::lexical_cast<std::string>(ID()));
+
+    bool all_bombard(true);
+    it = ships.begin();
+    while (it != ships.end()) {
+        if(!(*it)->CanBombard()) {
```
Comment (geoffthemedio) [inline universe/Fleet.cpp]: ...

File: default/stringtables/en.txt [hunk 6]
```diff
@@ -148,6 +148,24 @@ Fleet %1%
...
+
+NEW_TROOP_FLEET_NAME
+Troop Fleet %1%
+
+NEW_BOMBARD_FLEET_NAME
+Bomber Fleet %1%
+
+NEW_BATTLE_FLEET_NAME
+Battle Fleet %1%
+
```
Comment (geoffthemedio) [inline default/stringtables/en.txt:168]: Maybe also have a "Mixed Fleet" case, for use if all the ships aren't consistently anything?

Comment (LGM-Doyle) [inline default/stringtables/en.txt:168]: If it falls through everything, then the fleet name is just Fleet %1%

File: universe/Fleet.cpp (hunk 1 above)
Comment (LGM-Doyle) [inline universe/Fleet.cpp:1411]: No technical reason, but two style reasons.

First.

`for` implies that something should be done for all elements.

`while` implies that something (in this case a check for a ship type) should be done for some elements until something happens (find a specific type).  The reader expects the break statement.  

And second.

The code that I wanted to write was like:

```
`if (std::all_of(ship.begin(), ship.end(), [](const Ship & s){return s.IsMonster();}))
      return "Monster Fleet";`
```

`std::all_of` is only available in C++11 and it expands semantically into the while loop that I used.

**Comment by LGM-Doyle** [comment]
Thanks for the review.

Changes done and pushed.

**Pull request merged**